compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/repro-hoisting.expect.md

```diff
@@ -26,35 +26,29 @@ export const FIXTURE_ENTRYPOINT = {
 ```javascript
 import { c as _c } from "react/compiler-runtime";
 function Component(props) {
-  const $ = _c(4);
+  const $ = _c(3);
+  const wat = _temp;
+
+  const pathname_0 = props.wat;
+  const deeplinkItemId = pathname_0 ? props.itemID : null;
   let t0;
   if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
-    t0 = () => {};
+    t0 = () => wat();
```

Comment:
(Not feedback, just noticed and wanted to jot down a note)

This is slightly suboptimal as we could keep a sidemap of outlined functions (and also outline functions that only refer to already-outlined functions), but this might make more sense in terms of dx (e.g. if I were using a debugger, I would expect that a variable called `wat` exists)

Reply:
ohhh yeah, good idea. that would be a nice extension and it's probably still reasonably debuggable. for example, if we were to outline this function we could potentially outline it as:

```
function _temp2() {
  const wat = _temp;
  return wat();
}
```

to preserve the names for debugging.